src/pkg/packager/inspect.go

```diff
@@ -12,69 +12,68 @@ import (
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/pterm/pterm"
 	"oras.land/oras-go/v2/registry"
 )
 
 // Inspect list the contents of a package.
-func (p *Packager) Inspect(includeSBOM bool, outputSBOM string) error {
+func (p *Packager) Inspect(includeSBOM bool, outputSBOM string, validatePackage bool) error {
+	// Handle OCI packages that have been published to a registry
 	if utils.IsOCIURL(p.cfg.DeployOpts.PackagePath) {
-		spinner := message.NewProgressSpinner("Loading Zarf Package %s", p.cfg.DeployOpts.PackagePath)
-		ref, err := registry.ParseReference(strings.TrimPrefix(p.cfg.DeployOpts.PackagePath, "oci://"))
-		if err != nil {
-			return err
-		}
-
-		dst, err := utils.NewOrasRemote(ref)
-		if err != nil {
-			return err
-		}
-
-		// get the manifest
-		spinner.Updatef("Fetching the manifest for %s", p.cfg.DeployOpts.PackagePath)
-		layers, err := getLayers(dst)
-		if err != nil {
-			return err
-		}
-		spinner.Updatef("Loading Zarf Package %s", p.cfg.DeployOpts.PackagePath)
-		zarfYamlDesc := utils.Find(layers, func(d ocispec.Descriptor) bool {
-			return d.Annotations["org.opencontainers.image.title"] == "zarf.yaml"
-		})
-		err = pullLayer(dst, zarfYamlDesc, p.tmp.ZarfYaml)
-		if err != nil {
-			return err
-		}
 
-		if includeSBOM || outputSBOM != "" {
-			sbmomsTarDesc := utils.Find(layers, func(d ocispec.Descriptor) bool {
-				return d.Annotations["org.opencontainers.image.title"] == "sboms.tar"
-			})
-			err = pullLayer(dst, sbmomsTarDesc, p.tmp.SbomTar)
-			if err != nil {
+		// Download the full package if we are validating the package contents during the inspect
+		if validatePackage || p.cfg.DeployOpts.PublicKeyPath != "" {
+			if err := p.handleOciPackage(); err != nil {
 				return err
 			}
-			if err := archiver.Unarchive(p.tmp.SbomTar, filepath.Join(p.tmp.Base, "sboms")); err != nil {
+		} else {
+			pullSBOM := includeSBOM || outputSBOM != ""
+			if err := pullLayersForInspect(p.cfg.DeployOpts.PackagePath, p.tmp, pullSBOM); err != nil {
 				return err
 			}
 		}
-		err = utils.ReadYaml(p.tmp.ZarfYaml, &p.cfg.Pkg)
+		err := utils.ReadYaml(p.tmp.ZarfYaml, &p.cfg.Pkg)
 		if err != nil {
 			return err
 		}
-		spinner.Successf("Loaded Zarf Package %s", p.cfg.DeployOpts.PackagePath)
 	} else {
-		if err := p.loadZarfPkg(); err != nil {
-			return fmt.Errorf("unable to load the package: %w", err)
+		// This package exists on the local file system - extract the first layer of the tarball
+		if err := archiver.Unarchive(p.cfg.DeployOpts.PackagePath, p.tmp.Base); err != nil {
+			return fmt.Errorf("unable to extract the package: %w", err)
 		}
+		if err := p.readYaml(p.tmp.ZarfYaml, true); err != nil {
+			return fmt.Errorf("unable to read the zarf.yaml in %s: %w", p.tmp.Base, err)
+		}
+
 	}
 
 	pterm.Println()
 	pterm.Println()
 
 	utils.ColorPrintYAML(p.cfg.Pkg)
 
+	// Validate the package checksums and signatures if specified
+	if validatePackage || p.cfg.DeployOpts.PublicKeyPath != "" {
```

Comment:
This could maybe be one flag - if they provide a key attempt to validate, otherwise don't (having two is a little weird IMO)

Reply:
Having two is a bit weird, I was thinking the same thing but then I wasn't what to do about validating the checksums.

If we only have one flag, the checksums would either always get validate, or only get validated if the public key was provided.

I shied away from always validating the checksums because that ends up pulling a large amount of stuff when working with a remote package.

Reply:
Yeah we could make the public key path flag more explicit that you are both providing a public key and asking zarf to perform validation.

Reply:
Cleaned it up to be a bit more intuitive while also being more direct about warning the user for checksum/signature validation given the two scenarios of a remote OCI package and a local on-disk package.